crates/ott-common/src/discovery/dns.rs

```diff
@@ -21,6 +25,21 @@ pub struct DnsDiscoveryConfig {
     pub polling_interval: Option<Duration>,
 }
 
+fn deserialize_dns_server<'de, D>(deserializer: D) -> Result<Option<SocketAddr>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let mut buf = String::deserialize(deserializer)?;
+
+    if Option::is_none(&buf.find(':')) {
+        buf.push_str(":53");
+    }
+
+    Ok(Some(
+        SocketAddr::from_str(&buf).expect("Error: Could not unwrap"),
```

Comment:
You really shouldn't be calling expect here anyway, you are already returning a `Result`. `expect` triggers a panic, which is not practically recoverable. Panicking is reserved for instances where there is absolutely no possible way for the application to continue, and parsing failures do not fall into that category.

nit: "Could not unwrap" is too generic. Give context in error messages so they are actionable or searchable. Could not unwrap what? Could not parse what? Consider the user reading the error message.